velox/dwio/parquet/tests/ParquetTableScanTest.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/dwio/dwrf/test/utils/DataFiles.h"
+#include "velox/dwio/parquet/reader/ParquetReader.h"
+#include "velox/exec/tests/utils/HiveConnectorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/type/tests/FilterBuilder.h"
+#include "velox/type/tests/SubfieldFiltersBuilder.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::exec::test;
+
+static const std::string kWriter = "ParquetTableScanTest.Writer";
+
+class ParquetTableScanTest : public HiveConnectorTestBase {
+ protected:
+  using OperatorTestBase::assertQuery;
+
+  void SetUp() override {
+    HiveConnectorTestBase::SetUp();
+    parquet::registerParquetReaderFactory();
+  }
+
+  std::string getExampleFilePath(const std::string& fileName) {
+    return facebook::velox::test::getDataFilePath(
+        "velox/dwio/parquet/tests", "examples/" + fileName);
+  }
+
+  std::shared_ptr<connector::hive::HiveConnectorSplit> makeSplit(
```

Comment:
Can we extend `makeHiveConnectorSplit`  inside `velox/exec/tests/utils/HiveConnectorTestBase.h` and use it here?

Reply:
I don't see a good way of doing that, but I updated this method to re-use makeHiveConnectorSplit:

```
    auto split = makeHiveConnectorSplit(filePath);
    split->fileFormat = dwio::common::FileFormat::PARQUET;
    return split;
```